pkg/events/transport/kafka.go

```diff
@@ -5,4 +5,198 @@
 
 package transport
 
-// NO TESTS
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/Shopify/sarama"
+	"github.com/opentracing/opentracing-go"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/vmware/dispatch/pkg/events"
+	"github.com/vmware/dispatch/pkg/trace"
+)
+
+// Kafka Implements transport interface using Kafka broker.
+type Kafka struct {
+	producer     sarama.SyncProducer
+	consumer     sarama.Consumer
+	producerOnly bool
+}
+
+// OptKafkaSendOnly creates producer only. Subscribe operation will panic
+func OptKafkaSendOnly() func(k *Kafka) error {
+	return func(k *Kafka) error {
+		k.producerOnly = true
+		return nil
+	}
+}
+
+// NewKafka creates an instance of transport based on Kafka broker.
+func NewKafka(brokerAddrs []string, options ...func(k *Kafka) error) (*Kafka, error) {
+	syncProducer, err := sarama.NewSyncProducer(brokerAddrs, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	k := Kafka{
+		producer: syncProducer,
+	}
+	for _, option := range options {
+		// TODO: handle errors from options
+		option(&k)
+	}
+	if k.producerOnly {
+		return &k, nil
+	}
+
+	k.consumer, err = sarama.NewConsumer(brokerAddrs, nil)
+	return &k, err
+}
+
+// Publish publishes an event
+func (k *Kafka) Publish(ctx context.Context, event *events.CloudEvent, topic string, tenant string) error {
+	defer trace.Tracef("topic: %s", event.EventType)()
+	sp, _ := opentracing.StartSpanFromContext(
+		ctx,
+		"Kafka.Publish",
+	)
+	defer sp.Finish()
+
+	msg, err := fromEvent(event)
+	if err != nil {
+		return errors.Wrapf(err, "error when creating Kafka message from CloudEvent for topic %s", topic)
+	}
+	msg.Topic = topic
+
+	err = injectSpan(sp, msg)
```

Comment:
Maybe, it's not so simple :)

Reply:
Yeah you need some correlation data carrier, within the code that can be `context.Context`, you need some extra code to serialize that when sending over the wire.